x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2861 @@
+
+#### Via Bulk Actions
+
+Custom rules can currently be updated via the Rule Table's **Bulk Actions**, which uses the **Bulk Actions** - `POST /rules/_bulk_action` endpoint.
+
+Apart from enabling/disabling a rule - the only other action that modifies a rule's saved object and is currently possible for prebuilt rules - the user can use bulk actions to:
+
+- Add, delete and overwrite **index patterns**
+- Add, delete and overwrite **tags**
+- Add and overwrite **rule actions**
+- Update the rule's **schedule: interval and lookback time**
+- Select a **timeline template** for the rule
+
+As explained above, the UI validates that the five actions listed above are only possible to custom rules by using the `dryRun` mode of the **bulk actions** endpoint. If any of the selected rules in the request payload is a prebuilt rule, a message is displayed to the user informing them that modifying prebuilt rules is not possible, and that rule is removed from the subsequent request that is made in normal mode to actually edit the rule.
+
+**Expected behaviour for customizing prebuilt rules**
+
+- The Bulk Actions UI should now allow users to edit prebuilt rules.
+- All bulk actions actions listed above should now be applicable to prebuilt rules (This means that the changes in the API for the validation done in `dryRun` mode should not remove the rules from the payload anymore).
+
+> The `dryRun` mode request will still be necessary since we perform other checks that can still prevent prebuilt rules from being edited. For example, Machine Learning and ES|QL rules cannot be bulk edited to add or delete index patterns, since they don't have that field in their schema.
+
+#### Via the Rules Details Page
+
+The Rules Details page allows the user to modify the rule in two ways:
+
+- enable/disabling
+  - uses `POST /rules/_bulk_action` endpoint with enable/disable action
+- setting rule snoozing
+  - uses `POST /alerting/rule/{rule_id}/_snooze` endpoint
+
+Both of these actions are already possible for prebuilt rules, **so no changes are necessary here.**
+
+#### Via the Shared Exception Lists page
+
+The Shared Exception Lists page allows a user to create shared exception lists and then attach them to (or remove them from) a user's security rules.
+
+This page calls the **Patch Rule** - `PATCH /rules` endpoint to modify only the `exceptions_list` field of a rule saved object. Since a user can append an exception list to multiple rules via the UI, the endpoint is called multiple times simultaneously, one for each rule that is attached to or removed from.
+
+There is no validation regarding prebuilt rules in the UI for this use case: both the endpoint and the application currently allow the user to attach exceptions lists to both custom and prebuilt rules.
+
+This means that **no changes will be needed in this page.**
```

Comment:
Just want to double-check that we're on the same page regarding this: adding / editing / removing exceptions should not have any impact on `is_customized`. If `is_customized === true` or `is_customized === false`, it should stay `true` or `false` (correspondingly) in the following cases:

- a new shared exception list is added to the rule
- a shared exception list is removed from the rule
- an exception item is added to the rule's shared exception list
- a default exception list is created for the rule and an exception is added to it
- an exception item is removed from the rule's default exception list
- etc

Please document this expected behavior in the RFC.

Reply:
Yes, we're on the same page here. I'm detailing this behaviour in the `Calculating the is_customized field` section.